include/utils.h

```diff
@@ -5,44 +5,38 @@
 #include "assert.h"
 
 #ifdef _DEBUG
-constexpr bool debug_mode = true;
-constexpr bool assertion_mode = true;
-#else
-constexpr bool debug_mode = false;
-constexpr bool assertion_mode = false;
+#define _ASSERTIONS
 #endif
 
 #if defined(_ASSERTIONS) && !defined(_DEBUG)
-constexpr bool assertion_mode = true;
+bool assertion_mode = true;
 #endif
 
 #ifdef _LOGGING_LEVEL
-constexpr std::size_t global_log_level = _LOGGING_LEVEL;
+const std::size_t global_log_level = _LOGGING_LEVEL;
 #else
-constexpr std::size_t global_log_level = 0;
+const std::size_t global_log_level = 0;
 #endif
 
 inline void Assert(bool assertion)
 {
-  if constexpr (assertion_mode)
-  {
-    assert(assertion);
-  }
+#if defined(_ASSERTIONS)
```

Comment:
:-1 There is no point in having this flag, `assert` is already a macro that doesn't do anything if `NDEBUG` is set. If you want this to be a more SMT-switch specific thing its name should indicate that somehow.

Reply:
Yeah this is currently unused entirely. Eventually it could be a more complicated macro, but for now you're right that it's unneeded.